src/tiledb/cloud/notebook.py

```diff
@@ -181,37 +163,47 @@ def upload_notebook_contents(
     :return: TileDB array name, such as "tiledb://janedoe/testing-upload".
     """
 
-    if storage_credential_name is None:
-        storage_credential_name = (
-            tiledb.cloud.user_profile().default_s3_path_credentials_name
-        )
-    if storage_path is None:
-        storage_path = tiledb.cloud.user_profile().default_s3_path
+    storage_credential_name = storage_credential_name or (
+        client.default_user().default_s3_path_credentials_name
+    )
+    storage_path = storage_path or client.default_user().default_s3_path
+    namespace = namespace or client.default_user().username
 
-    if storage_credential_name is None:
+    if not storage_credential_name:
         raise tiledb_cloud_error.TileDBCloudError(
-            f"No storage credentials found in account. Please add them there, or pass them in explicitly here."
-        ) from e
-    if storage_path is None:
+            "No storage credentials found in account."
+            " Please add them there, or pass them in explicitly here."
+        )
+    if not storage_path:
         raise tiledb_cloud_error.TileDBCloudError(
-            f"No storage path found in account. Please add it there, or pass it in explicitly here."
-        ) from e
+            "No storage path found in account."
+            " Please add it there, or pass it in explicitly here."
+        )
+    if not namespace:
+        raise tiledb_cloud_error.TileDBCloudError("Namespace to store notebook unset.")
 
-    ctx = tiledb.cloud.Ctx(
-        {"rest.creation_access_credentials_name": storage_credential_name}
-    )
+    ctx = client.Ctx({"rest.creation_access_credentials_name": storage_credential_name})
 
-    if on_exists is OnExists.FAIL:
+    tiledb_uri = f"tiledb://{namespace}/{array_name}"
+    try:
+        arr = tiledb.open(tiledb_uri, ctx=ctx)
```

Comment:
You can also do `tiledb.array_exists`, which is light weight check compared to a full array open.

Reply:
It looks like internally it does the same thing, but it does not (currently) accept a `ctx`.

Reply:
It does not quite do the same thing. Opening the array might also return the nonempty domain, metadata, and fragment metadata in addition to the array schema. The exist function today will only fetch the schema and then discard it.

Reply:
https://github.com/TileDB-Inc/TileDB-Py/blob/0.21.1/tiledb/highlevel.py#L139-L153